operator/pkg/name/name.go

```diff
@@ -74,6 +78,13 @@ var (
 		NodeAgentComponentName,
 		CNIComponentName,
 	}
+	AllAddonComponentNames = []ComponentName{
```

Comment:
To align with other components grouup's set, it may be better to use a map here

Reply:
I added the map below and also a respective `IsAddonComponent()` method, which I now use to determine the correct spec paths; this is correct as the old `!IsAddonComponent()` would also trigger for deprecated components